### PR TITLE
Update hostNetwork docs

### LINKED ISSUE #2114: When using hostNetwork: true for envoy the dnsPolicy should also be set
**What steps did you take and what happened:**
When using the example deployment and setting hostNetwork: true envoy can't communicate with contour due to DNS

**What did you expect to happen:**
Envoy able to talk to Contour

**Anything else you would like to add:**
This is because the dnsPolicy needs to be explicitly set to ClusterFirstWithHostNet when using hostNetwork: true.  After setting this the issue is resolved

**Environment:**

- Contour version: v1.1.0
- Kubernetes version: (use `kubectl version`):
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.5", GitCommit:"20c265fef0741dd71a66480e35bd69f18351daea", GitTreeState:"clean", BuildDate:"2019-10-15T19:16:51Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.6", GitCommit:"96fac5cd13a5dc064f7d9f4f23030a6aeface6cc", GitTreeState:"clean", BuildDate:"2019-08-19T11:05:16Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}
- Kubernetes installer & version:
Rancher v2.3.1
- Cloud provider or hardware configuration:
VMWare virtual servers

### PR DESCRIPTION
**Pull request by dbason**
Fixes #2114

### CODE REVIEW

**Comment by stevesloka** [comment]
Why does this need changed? I've never had a problem with dns. Is this k8s distro specific?

**Comment by stevesloka** [comment]
We shouldn't update docs in the past either I don't think. V1.1 is the latest version.

**Comment by dbason** [comment]
@stevesloka this is generic - k8s docs https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-dns-policy state that for hostNetwork the DNS Policy needs to be explicitly changed.  This is stated in the README for the example deployment, so this is just updating the online docs to match.

I'll revert the second commit now and just update v1.1

**Comment by stevesloka** [comment]
I'll take a look, thanks for the link. Can you update master and v1.1 docs?

**Comment by dbason** [comment]
Done :)

File: site/docs/master/deploy-options.md [hunk 1]
```diff
@@ -200,7 +200,7 @@ Now you can point your browser at the specified port on any node in your cluster
 
 You can run Contour without a Kubernetes Service at all.
 This is done by having the Contour pod run with host networking.
-Do this with `hostNetwork: true` on your pod definition.
+Do this with `hostNetwork: true` and `dnsPolicy: ClusterFirstWithHostNet` on your pod definition.
 Envoy will listen directly on port 8080 on each host that it is running.
```
Comment (stevesloka) [inline site/docs/master/deploy-options.md]: Could you update the port here to `80` from `8080`?

File: site/docs/master/deploy-options.md [hunk 2]
```diff
@@ -200,7 +200,7 @@ Now you can point your browser at the specified port on any node in your cluster
...
 This is best paired with a DaemonSet (perhaps paired with Node affinity) to ensure that a single instance of Contour runs on each Node.
```
Comment (stevesloka) [inline site/docs/master/deploy-options.md]: Contour runs as a deployment and Envoy runs as a daemonset. Could you update this to reflect the changes from the old deployment model.

**Pull request merged**